File: EcoriTechTest.Domain.Model/Entities/Character.cs
using EcoriTechTest.Domain.Model.Interfaces;
using EcoriTechTest.Domain.Model.ValueObjects;

namespace EcoriTechTest.Domain.Model.Entities
{
    public class Character : IEntity
    {
        public int Id { get; set; }

        public string Name { get; set; }

        public string Description { get; set; }

        public DateTimeOffset Modified { get; set; }

        public string ResourceURI { get; set; }

        public List<UrlValue> Urls { get; set; }

        public Image Thumbnail { get; set; }

        public ComicList Comics { get; set; }

        public StoryList Stories { get; set; }

        public EventList Events { get; set; }

        public SeriesList Series { get; set; }
    }
}

File: EcoriTechTest.WebApp/src/app/character/models/character.ts
import { ComicList } from "./value-object/comic-list";
import { EventList } from "./value-object/event-list";
import { Image } from "./value-object/image";
import { SeriesList } from "./value-object/series-list";
import { StoryList } from "./value-object/story-list";
import { UrlValue } from "./value-object/url-value";

export class Character {
  id: number = 0;
  name: string = '';
  description: string = '';
  modified: Date = new Date();
  resourceURI: string = '';
  urls: UrlValue[] = [];
  thumbnail: Image = new Image();
  comics: ComicList = new ComicList();
  stories: StoryList = new StoryList();
  events: EventList = new EventList();
  series: SeriesList = new SeriesList();
}
